routing/index_graph_starter.hpp

```diff
@@ -16,18 +23,25 @@ namespace routing
 class IndexGraphStarter final
 {
 public:
+  friend struct routing_test::RestrictionTest;
```

Comment:
Move this to the private section, please. friends declarations are useless for almost all readers of the class API, moreover, when the friend declaration appears in the private section it seems for me as "Okay, this is a private section, and here is the friend that can call all this stuff".